pam/integration-tests/native_test.go

```diff
@@ -0,0 +1,215 @@
+		"Authenticate user with qr code":                              {tape: "qr_code"},
+		"Authenticate user and reset password while enforcing policy": {tape: "mandatory_password_reset"},
+		"Authenticate user and reset password with mismatch":          {tape: "mandatory_password_reset_with_mismatch"},
+		"Authenticate user and offer password reset":                  {tape: "optional_password_reset_skip"},
+		"Authenticate user and accept password reset":                 {tape: "optional_password_reset_accept"},
+		"Authenticate user switching auth mode":                       {tape: "switch_auth_mode"},
+		"Authenticate user switching username":                        {tape: "switch_username"},
+		"Authenticate user switching to local broker":                 {tape: "switch_local_broker"},
+		"Authenticate user and add it to local group":                 {tape: "local_group"},
+		"Authenticate with warnings on unsupported arguments":         {tape: "simple_auth_with_unsupported_args"},
+
+		"Remember last successful broker and mode":      {tape: "remember_broker_and_mode"},
+		"Autoselect local broker for local user":        {tape: "local_user"},
+		"Autoselect local broker for local user preset": {tape: "local_user_preset"},
+
+		"Deny authentication if current user is not considered as root": {tape: "not_root", currentUserNotRoot: true},
+
+		"Deny authentication if max attempts reached":                         {tape: "max_attempts"},
+		"Deny authentication if user does not exist":                          {tape: "unexistent_user"},
+		"Deny authentication if usernames dont match":                         {tape: "mismatch_username"},
+		"Deny authentication if newpassword does not match required criteria": {tape: "bad_password"},
+
+		"Exit authd if local broker is selected": {tape: "local_broker"},
+		// FIXME: While this works now, it requires proper handling via signal_fd
+		"Exit authd if user sigints": {tape: "sigint"},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			outDir := t.TempDir()
+			err := os.Symlink(filepath.Join(clientPath, "pam_authd"),
+				filepath.Join(outDir, "pam_authd"))
+			require.NoError(t, err, "Setup: symlinking the pam client")
+
+			cliLog := prepareCLILogging(t)
+			t.Cleanup(func() {
+				saveArtifactsForDebug(t, []string{
+					filepath.Join(outDir, tc.tape+".gif"),
+					filepath.Join(outDir, tc.tape+".txt"),
+					cliLog,
+				})
+			})
+
+			gpasswdOutput := filepath.Join(outDir, "gpasswd.output")
+			groupsFile := filepath.Join(testutils.TestFamilyPath(t), "gpasswd.group")
+			socketPath := runAuthd(t, gpasswdOutput, groupsFile, !tc.currentUserNotRoot)
+
+			const socketPathEnv = "AUTHD_TESTS_CLI_AUTHENTICATE_TESTS_SOCK"
+			// #nosec:G204 - we control the command arguments in tests
+			cmd := exec.Command("env", "vhs", filepath.Join(currentDir, "testdata", "native-tapes", tc.tape+".tape"))
```

Comment:
I find it weird having both `tapes` and `native-tapes` directories. Maybe move the `native-tapes` into `tapes/native` and fix the other fixtures to follow the same pattern? (e.g. `tapes/cli`) 

Not sure if the `cli` name is the best one, but I don't have a better idea right now so if you find something better we can use it instead.

Reply:
Ok, to reduce the diff maybe we can do it in another PR (for the others?)

Reply:
The diff should only point out that the files were renamed so it won't interfere too much with the diff amount. Besides, it's already 16k, so it won't change much :joy: